### ISSUE TITLE
riff delete --all doesn't work if not in function-dir (used with -n)

### ISSUE THREAD

**Issue by ericbottard** [projectriff/riff#338]
I assume it tries to read the local `yaml` file. It should query the function resource instead

**Comment by dturanski** [comment]
True, e.g., 
```
$cd ~/go/src/github.com/projectriff/riff/riff-cli/test_data/shell/echo 
$ riff delete --all --dry-run -n echo

Delete Command: kubectl delete --namespace default -f .../go/src/github.com/projectriff/riff/riff-cli/test_data/shell/echo/echo-function.yaml -f .../go/src/github.com/projectriff/riff/riff-cli/test_data/shell/echo/echo-topics.yaml
```
This normally works according to `kubectl` docs. Can you provide more details?

**Comment by ericbottard** [comment]
I wrote "if NOT in function dir".

eg

``` 
~/Documents/projects/go/src/github.com/projectriff/riff   master  riff delete --dry-run --all -n wordcount

Delete Command: kubectl delete --namespace default

 ~/Documents/projects/go/src/github.com/projectriff/riff   master  riff delete --all -n wordcount
Deleting . resources

exit status 1: error: You must provide one or more resources by argument or filename.
Example resource specifications include:
   '-f rsrc.yaml'
   '--filename=rsrc.json'
   '<resource> <name>'
   '<resource>'

exit status 1
```

**Comment by dturanski** [comment]
My bad. I saw "in function dir"

**Comment by trisberg** [comment]
For `--all` you need to provide the `-f` option for the directory that contains the yaml files, unless you are in that directory already. It won't work by providing the `-n` options since we don't know the name of the associated topics. We don't try to interrogate the yaml content of the function ATM.

**Comment by ericbottard** [comment]
I would argue we should either do it (interrogate k8s) or fail if `--all` is used without `-f` then

**Comment by trisberg** [comment]
looks like Dave has a PR #341  - it worked for me, although I'm currently on a flight layover so won't have time to merge it.

**Comment by trisberg** [comment]
Fixed via #341
